### PR TITLE
Added NumPy take method

### PR DESCRIPTION
**Pull request by joshsalce**
take https://github.com/unifyai/ivy/issues/19656

Note: There is no Ivy Experimental API implementation, so I implemented it using Ivy array methods and Python. It will have to be simplified once implemented in Ivy

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 1]
```diff
@@ -178,3 +178,43 @@ def test_numpy_compress(
...
+        indices_dtypes=["int32", "int64"],
+        min_num_dims=1,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=10,
+        indices_same_dims=True,
+        valid_bounds=False,
+    ),
+    mode=st.sampled_from(["clip", "wrap", "raise"]),
+    test_with_out=st.just(False),
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:196]: Seems to be an issue with this handle in testing [here](https://github.com/unifyai/ivy/actions/runs/5615801049/job/15216863669?pr=19725#step:3:220).

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Seems to be an issue with your test. Maybe just check over and make sure that all your args are right for the test functions

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py [hunk 2]
```diff
@@ -178,3 +178,45 @@ def test_numpy_compress(
...
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=10,
+        indices_same_dims=True,
+        valid_bounds=False,
+    ),
+    mode=st.sampled_from(["clip", "wrap", "raise"]),
+    test_with_out=st.just(False)
+)
+def test_numpy_take(
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py:199]: Seems to be an issue with arguments according to tests [here](https://github.com/unifyai/ivy/actions/runs/5659074707/job/15331825717?pr=19725#step:3:182). Would double check that all of your args and kwargs are correct :)

**Review by jkeane508 — CHANGES_REQUESTED** [review]

**Comment by DecFox** [comment]
Hey @joshsalce, thanks for the PR. Sorry for the late feedback. I left some comments which should hopefully help with the tests.

**Review by DecFox — CHANGES_REQUESTED** [review]

**Comment by Ishticode** [comment]
closing this as take is already implemented in numpy frontend https://github.com/unifyai/ivy/blob/ea402610b7e2cdf77eccf2460e7b6be42e71c2b0/ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py#L88

**Pull request closed without merging**